From 185626579f894ba5c3b3a1ecd290435bb914f03b Mon Sep 17 00:00:00 2001
From: Sylvain Pichon <sylvainx.pichon@intel.com>
Date: Tue, 19 Aug 2014 17:03:05 +0200
Subject: [PATCH] [REVERT ME]Fix for legacy policy usage

BZ: 217971

New L-dessert audio policy is used by default.

This patch makes some corrections so that code
can build when USE_AUDIO_POLICY_LEGACY flag is
used.

Change-Id: I8042956288b6bb5a107aa8ecce968e9d1dce82b4
Signed-off-by: Sylvain Pichon <sylvainx.pichon@intel.com>
---
 services/audiopolicy/Android.mk                         | 1 +
 services/audiopolicy/AudioPolicyClientImplLegacy.cpp    | 4 ++--
 services/audiopolicy/AudioPolicyInterfaceImplLegacy.cpp | 1 +
 3 files changed, 4 insertions(+), 2 deletions(-)

diff --git a/services/audiopolicy/Android.mk b/services/audiopolicy/Android.mk
index 6512c38..cab1bb4 100644
--- a/services/audiopolicy/Android.mk
+++ b/services/audiopolicy/Android.mk
@@ -20,6 +20,7 @@ endif
 
 LOCAL_C_INCLUDES := \
     $(TOPDIR)frameworks/av/services/audioflinger \
+    $(TOPDIR)frameworks/av/include/media \
     $(call include-path-for, audio-effects) \
     $(call include-path-for, audio-utils)
 
diff --git a/services/audiopolicy/AudioPolicyClientImplLegacy.cpp b/services/audiopolicy/AudioPolicyClientImplLegacy.cpp
index 9639096..fe054a9 100644
--- a/services/audiopolicy/AudioPolicyClientImplLegacy.cpp
+++ b/services/audiopolicy/AudioPolicyClientImplLegacy.cpp
@@ -69,7 +69,7 @@ static audio_io_handle_t open_output(audio_module_handle_t module,
                                     audio_channel_mask_t *pChannelMask,
                                     uint32_t *pLatencyMs,
                                     audio_output_flags_t flags,
-                                    const audio_offload_info_t *offloadInfo)
+                                    audio_offload_info_t *offloadInfo)
 {
     sp<IAudioFlinger> af = AudioSystem::get_audio_flinger();
     if (af == 0) {
@@ -123,7 +123,7 @@ audio_io_handle_t aps_open_output_on_module(void *service __unused,
                                                    audio_channel_mask_t *pChannelMask,
                                                    uint32_t *pLatencyMs,
                                                    audio_output_flags_t flags,
-                                                   const audio_offload_info_t *offloadInfo)
+                                                   audio_offload_info_t *offloadInfo)
 {
     return open_output(module, pDevices, pSamplingRate, pFormat, pChannelMask,
                           pLatencyMs, flags, offloadInfo);
diff --git a/services/audiopolicy/AudioPolicyInterfaceImplLegacy.cpp b/services/audiopolicy/AudioPolicyInterfaceImplLegacy.cpp
index f20c070..1913ce9 100644
--- a/services/audiopolicy/AudioPolicyInterfaceImplLegacy.cpp
+++ b/services/audiopolicy/AudioPolicyInterfaceImplLegacy.cpp
@@ -24,6 +24,7 @@
 #include <system/audio.h>
 #include <system/audio_policy.h>
 #include <hardware/audio_policy.h>
+#include <AudioPolicyHelper.h>
 
 namespace android {
 
-- 
1.9.2

